### PR TITLE
feat(generate): create parent directories required for blueprints if they do not exist

### LINKED ISSUE #3307: Cannot read property 'kind' of undefined
### Versions.
ng version
angular-cli: 1.0.0-beta.21
node: 7.1.0
os: linux x64

### Repro steps.
ng g component components/beat-feed
 
### The log given by the failure.
ng g component components/beat-feed
installing component
  create src/app/components/beat-feed/beat-feed.component.scss
  create src/app/components/beat-feed/beat-feed.component.html
  create src/app/components/beat-feed/beat-feed.component.spec.ts
  create src/app/components/beat-feed/beat-feed.component.ts
Cannot read property 'kind' of undefined
TypeError: Cannot read property 'kind' of undefined
    at visit (/home/user/Projects/prwire/tools/@angular/tsc-wrapped/src/symbols.ts:34:54)
    at visitEachNode (/home/user/Projects/prwire/interface/node_modules/typescript/lib/typescript.js:12983:30)
    at Object.forEachChild (/home/user/Projects/prwire/interface/node_modules/typescript/lib/typescript.js:13143:24)
    at Symbols.buildImports (/home/user/Projects/prwire/tools/@angular/tsc-wrapped/src/symbols.ts:100:10)
    at Symbols.get [as symbols] (/home/user/Projects/prwire/tools/@angular/tsc-wrapped/src/symbols.ts:21:12)
    at Symbols.resolve (/home/user/Projects/prwire/tools/@angular/tsc-wrapped/src/symbols.ts:10:63)
    at /home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/src/ast-utils.js:104:36
    at FilterSubscriber._next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/filter.ts:86:31)
    at FilterSubscriber.Subscriber.next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Subscriber.ts:95:12)
    at MapSubscriber._next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/map.ts:84:22)
    at MapSubscriber.Subscriber.next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Subscriber.ts:95:12)
    at FilterSubscriber._next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/filter.ts:92:24)
    at FilterSubscriber.Subscriber.next (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Subscriber.ts:95:12)
    at ReplaySubject._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/ReplaySubject.ts:41:18)
    at ReplaySubject.Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:95:21)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:95:21)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at FilterOperator.call (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/filter.ts:61:19)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:93:16)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at MapOperator.call (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/map.ts:54:19)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:93:16)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at FilterOperator.call (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/filter.ts:61:19)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:93:16)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at FilterOperator.call (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/filter.ts:61:19)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:93:16)
    at Observable._subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:155:24)
    at MapOperator.call (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/map.ts:54:19)
    at Observable.subscribe (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/Observable.ts:93:16)
    at /home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/toPromise.ts:24:10
    at Observable.toPromise (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/node_modules/rxjs/src/operator/toPromise.ts:22:10)
    at _addSymbolToNgModuleMetadata (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/src/ast-utils.js:135:10)
    at Object.addDeclarationToModule (/home/user/Projects/prwire/interface/node_modules/@angular-cli/ast-tools/src/ast-utils.js:230:12)
    at Class.afterInstall (/home/user/Projects/prwire/interface/node_modules/angular-cli/blueprints/component/index.js:147:18)
    at tryCatch (/home/user/Projects/prwire/interface/node_modules/rsvp/dist/lib/rsvp/-internal.js:215:12)
    at invokeCallback (/home/user/Projects/prwire/interface/node_modules/rsvp/dist/lib/rsvp/-internal.js:230:13)
    at publish (/home/user/Projects/prwire/interface/node_modules/rsvp/dist/lib/rsvp/-internal.js:198:7)
    at flush (/home/user/Projects/prwire/interface/node_modules/rsvp/dist/lib/rsvp/asap.js:85:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

### PR DESCRIPTION
**Pull request by Meligy**
fixes #3307

### CODE REVIEW

File: tests/acceptance/generate-component.spec.js [hunk 1]
```diff
@@ -178,6 +178,20 @@ describe('Acceptance: ng generate component', function () {
       });
   });
 
+  it(`nonExistingDir${path.sep}myComp will create dir and succeed`, () => {
+    const testPath =
+      path.join(root, 'tmp', 'foo', 'src', 'app', 'non-existing-dir', 'my-comp', 'my-comp.component.ts');
+    const appModule = path.join(root, 'tmp', 'foo', 'src', 'app', 'app.module.ts');
+    return ng(['generate', 'component', `non-existing-dir${path.sep}myComp`])
```
Comment (filipesilva) [inline tests/acceptance/generate-component.spec.js]: Can you test `non/existing/dir/myComp` instead? It should be fine to use `/` everywhere, and I'd also like to be sure that nested dirs work.

**Comment by filipesilva** [comment]
Adding @Brocco as a reviewer as well. I added my review but he's the authority on this topic.

File: tests/acceptance/generate-component.spec.js (hunk 1 above)
Comment (Meligy) [inline tests/acceptance/generate-component.spec.js]: Done.

**Comment by filipesilva** [comment]
@Brocco are you happy with the tests provided?

**Comment by Brocco** [comment]
I just changed my review and marked this as approved.

**Pull request closed without merging**